File: 刷题/cpp/56. 合并区间.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int> > ans;
        sort(intervals.begin(),intervals.end());
        int left=0,right=0;
        for(int i=0;i<intervals.size();i++){
            left=intervals[i][0];
            right=intervals[i][1];
            if (ans.size()==0 || ans.back()[1]<left){
                ans.push_back({left,right});
            }else{
                right=max(right,ans.back()[1]);
                ans.back()[1]=right;
            }
        }
        return ans;
    }
};

File: 刷题/golang/56. 合并区间.go
func merge(intervals [][]int) [][]int {
    sort.Slice(intervals, func(i,j int) bool {
        return intervals[i][0]<intervals[j][0];
    })
    left,right:=0,0
    var ans [][]int
    for _,interval := range intervals{
        left=interval[0]
        right=interval[1]
        if len(ans)==0 || ans[len(ans)-1][1]<left {
            ans=append(ans,append([]int(nil),[]int{left,right}...))
        } else {
            right=max(right,ans[len(ans)-1][1])
            ans[len(ans)-1][1]=right
        }
    } 
    return ans
}
func max(x,y int) int{
    if x>y {
        return x
    }
    return y
}